itou/communications/forms.py

```diff
@@ -35,12 +32,12 @@ class Meta:
 
     def save(self, commit=True):
         if "image" in self.changed_data:
-            extension = pathlib.Path(self.instance.image.name).suffix
-            self.instance.image.name = f"{uuid.uuid4()}{extension}"
-            if image := self.initial.get("image"):
-                File.objects.filter(key=image.name).update(deleted_at=timezone.now())
-        instance = super().save(commit=commit)
-        if "image" in self.changed_data:
-            instance.image_storage = File.objects.create(key=instance.image.name)
-            instance.save()
-        return instance
+            if image_storage := self.instance.image_storage:
+                image_storage.deleted_at = timezone.now()
+                image_storage.save(update_fields=["deleted_at"])
+
+            self.instance.image_storage = File.objects.create(
+                key_prefix="news-images/", filename=self.cleaned_data["image"].name
+            )
```

Comment:
Cela va créer une désynchro entre les objets "File" et les fichiers dans S3.
Pour tester, tu peux:
- créer une campagne avec un `AnnouncementItem` ayant une image
- lancer `./manage.py sync_s3_files`
- venir éditer l'image de l'`AnnouncementItem` dans l'admin
- relancer `./manage.py sync_s3_files`

Cela devrait te dire qu'il a trouvé un nouvel objet dans S3:
`Completed bucket sync: found permanent=1 and temporary=0 files in the bucket`
et qu'il a aussi trouvé un `File` en DB qui ne correspond à rien dans S3:
`1 database files do not exist in the bucket: ['news-images/25cd98f2-03ae-44aa-ba99-1331cbfe8a8a.gif']`

Reply:
`sync_s3_files` prendra sa retraite avec #6470